src/dotnet/ReSharperPlugin.FluentAssertions/Psi/FluentAssertionsPredefinedType.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using JetBrains.Metadata.Reader.API;
+using JetBrains.Metadata.Reader.Impl;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.Modules;
+
+namespace ReSharperPlugin.FluentAssertions.Psi
+{
+    public class FluentAssertionsPredefinedType
+    {
+        private static readonly Dictionary<IClrTypeName, int> s_typeNameIndex;
+
+        public static readonly IClrTypeName ASSERT_FQN =
+            new ClrTypeName(nameof(NUnit) + "." + nameof(NUnit.Framework) + "." + nameof(NUnit.Framework.Assert));
+
+        static FluentAssertionsPredefinedType()
+        {
+            // Collect predefined type names through reflection
+            s_typeNameIndex = new Dictionary<IClrTypeName, int>();
+
+            foreach (var info in typeof(FluentAssertionsPredefinedType).GetFields())
+            {
+                if (info.IsStatic && typeof(IClrTypeName).IsAssignableFrom(info.FieldType))
+                {
+                    var clrTypeName = (IClrTypeName) info.GetValue(obj: null);
+                    s_typeNameIndex.Add(clrTypeName, s_typeNameIndex.Count);
+                }
+            }
```

Comment:
To be honest, I don't get why do we need this complicated system which stores indices of static fields in a dictionary.